app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,49 @@
+class TasksController < ApplicationController
+  before_action :find_task, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @tasks = Task.all
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      redirect_to @task
+    else
+      render :new
+    end
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to @task
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @task.destroy
+      redirect_to tasks_path
+    end
+  end
+
+  private
+  def find_task
+    @task = Task.find(params[:id])
```

Comment:
Can raise ActiveRecord::RecordNotFound, so good practice is to handle situation when we are trying to update/destroy/edit task that already was deleted.